src/lib.rs

```diff
@@ -260,23 +287,54 @@ impl<'de> Deserialize<'de> for Message {
     }
 }
 
+#[cfg_attr(feature = "typed-builder", derive(typed_builder::TypedBuilder))]
+#[cfg_attr(feature = "typed-builder", builder(doc))]
+#[derive(Default)]
+/// Verification options and configuration
+pub struct VerificationOpts {
```

Comment:
I wonder if we should call this `ValidationOpts` instead of `VerificationOpts`, to signify that this is about validating the content of the siwe message rather than verifying the signature. Of course then the rpc provider wouldn't belong here. Thoughts?

Reply:
The vocabulary we agreed on across SIWE libraries is to say verification for the verification of a signed/existing message (including all checks), and validation for the construction of a message. Of course validation doesn't appear here as we can encode all constraints in the types (although now that I think about it we are missing certain constraints, like the nonce length, maybe something to fix).